src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -363,15 +363,15 @@ private ProgrammingExerciseStudentParticipation copyRepository(ProgrammingExerci
         // only execute this step if it has not yet been completed yet or if the repository url is missing for some reason
         if (!participation.getInitializationState().hasCompletedState(InitializationState.REPO_COPIED) || participation.getVcsRepositoryUrl() == null) {
             final var projectKey = programmingExercise.getProjectKey();
-            final var participantIdentifier = participation.getParticipantIdentifier() + (participation.isTestRun() ? "-practice" : "");
+            final var repoName = (participation.isTestRun() ? "practice-" : "") + participation.getParticipantIdentifier();
```

Comment:
Add a util method for this formatting of the repoName/buildPlan that's repeated in 3 different places